src/test/java/edu/carroll/cs389application/service/ImageServiceImplTest.java

```diff
@@ -0,0 +1,287 @@
+package edu.carroll.cs389application.service;
+
+import edu.carroll.cs389application.jpa.model.Login;
+import edu.carroll.cs389application.jpa.model.UserImage;
+import edu.carroll.cs389application.service.ImageServiceImpl.ErrorCode;
+import edu.carroll.cs389application.web.form.ImageForm;
+import edu.carroll.cs389application.web.form.LoginForm;
+import jakarta.validation.constraints.AssertTrue;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.MockitoAnnotations;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.data.util.Pair;
+import org.springframework.mock.web.MockMultipartFile;
+import org.springframework.web.multipart.MultipartFile;
+
+import javax.imageio.ImageIO;
+import java.awt.image.BufferedImage;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.List;
+
+import static org.junit.Assert.*;
+
+/**
+ * Unit tests for the ImageService implementation.
+ * <p>
+ * Combined pullimages and saveimage into one because they test all the functionality together.
+ */
+@SpringBootTest
+public class ImageServiceImplTest {
+
+    @Autowired
+    private ImageService imageService;
+
+    @Autowired
+    private UserService userService;
+
+    private final String testuser = "testuser";
+
+    private final LoginForm testform = new LoginForm(testuser);
+
+    /**
+     * Set up the mock objects before each test case.
+     *
+     * @throws IOException if there is an error in the setup process.
+     */
+    @BeforeEach
+    public void setup() throws IOException {
+        MockitoAnnotations.openMocks(this);
+        if (userService.validateUsername(testform)) {
+            //Create and validate our user in the DB.
+        }
+    }
+
+    /**
+     * This method will clean up the database after writing images.
+     *
+     * @throws IOException will throw if images failed to clean
+     */
+    @AfterEach
+    public void cleanup() throws IOException {
+        Login user = userService.loginFromUsername(testuser);
```

Comment:
FYI, if you added the `@Transactional` annotation just above the @SpringBootTest annotation, you could remove this method.